src/psyclone/transformations.py

```diff
@@ -563,98 +562,232 @@ def apply(self, node, options=None):
         return rval
 
 
-class OMPLoopTrans(ParallelLoopTrans):
+class OMPTargetTrans(RegionTrans):
+    '''
+    Adds an OpenMP target directive to a region of code.
+
+    For example:
+
+    >>> from psyclone.psyir.frontend.fortran import FortranReader
+    >>> from psyclone.psyir.nodes import Loop
+    >>> from psyclone.transformations import OMPTargetTrans
+    >>>
+    >>> tree = FortranReader().psyir_from_source("""
+    >>>     subroutine my_subroutine()
+    >>>         integer, dimension(10, 10) :: A
+    >>>         integer :: i
+    >>>         integer :: j
+    >>>         do i = 1, 10
+    >>>             do j = 1, 10
+    >>>                 A(i, j) = 0
+    >>>             end do
+    >>>         end do
+    >>>     end subroutine
+    >>>     """
+    >>> omptargettrans = OMPTargetTrans()
+    >>> omptargettrans.apply(tree.walk(Loop))
+
+    will generate:
+
+    .. code-block:: fortran
+
+        subroutine my_subroutine()
+            integer, dimension(10, 10) :: A
+            integer :: i
+            integer :: j
+            $!omp target
```

Comment:
s/$!/!$/ and again for the closing directive :-)

Reply:
Oops, fixed